src/System.IO.IsolatedStorage/tests/System/IO/IsolatedStorage/CreateFileTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.IsolatedStorage
+{
+    public class CreateFileTests : IsoStorageTest
+    {
+        [Fact]
+        public void CreateFile_ThrowsArugmentNull()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                Assert.Throws<ArgumentNullException>(() => isf.CreateFile(null));
+            }
+        }
+
+        [Fact]
+        public void CreateFile_ThrowsIsolatedStorageException()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                isf.Remove();
+                Assert.Throws<IsolatedStorageException>(() => isf.CreateFile("foo"));
+            }
+        }
+
+        [Fact]
+        public void CreateFile_ThrowsObjectDisposed()
+        {
+            IsolatedStorageFile isf;
+            using (isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+            }
+
+            Assert.Throws<ObjectDisposedException>(() => isf.CreateFile("foo"));
+        }
+
+        [Fact]
+        public void CreateFile_ThrowsInvalidOperationException()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                isf.Close();
+                Assert.Throws<InvalidOperationException>(() => isf.CreateFile("foo"));
+            }
+        }
+
+        [Fact]
+        public void CreateFile_RaisesArgumentException()
+        {
+            using (IsolatedStorageFile isf = IsolatedStorageFile.GetUserStoreForAssembly())
+            {
+                Assert.Throws<ArgumentException>(() => isf.CreateFile("\0bad"));
```

Comment:
Null is the only one that doesn't vary per platform. This is more about raising the exception, not completeness of the underlying functionality.